Wb store id option

wbobillet opened the pull request.

wbobillet committed:
impletemtation of the PersistID option

wbobillet committed:
remove some warnings

hendrik-skubch commented on alica_engine/src/engine/teammanager/TeamManager.cpp:
```diff
@@ -84,12 +84,24 @@ void TeamManager::readSelfFromConfig()
-    int id = sc["Local"]->tryGet<int>(-1, "Local", "ID", NULL);
-    if (id != -1) {
+    
+    constexpr auto notAValidID = (unsigned long long)(-1);
```
uint64_t　is the cleaner choice imo.
also c-style casts?
I'd recommend clang-tidy 
/ramble off ;)

abhi1287 commented on alica_engine/src/engine/teammanager/TeamManager.cpp:
```diff
@@ -84,12 +84,24 @@ void TeamManager::readSelfFromConfig()
     } else {
         _localAnnouncement.senderID = _engine->generateId(DEFAULT_AGENT_ID_SIZE);
         ALICA_DEBUG_MSG("tm: Auto generated id " << _localAnnouncement.senderID);
+        bool persist_id = sc["Alica"]->tryGet<bool>(false, "Alica", "PersistID", NULL);
```
persistId by naming convention.

abhi1287 commented on alica_engine/src/engine/teammanager/TeamManager.cpp:
```diff
@@ -84,12 +84,24 @@ void TeamManager::readSelfFromConfig()
-    int id = sc["Local"]->tryGet<int>(-1, "Local", "ID", NULL);
-    if (id != -1) {
+    
+    constexpr auto notAValidID = (unsigned long long)(-1);
```
Can use ULLONG_MAX

hendrik-skubch replied:
better yet std::numeric_limits<uint64_t>::max();

veerajsk committed:
Casting, id type, naming & missing include fixes

1. Change unsigned long long to uint64_t
2. Use numeric_limits<uint64_t>::max() to get the max value
3. Use static_cast instead of c-style cast to invoke conversion function
4. Change persist_id to persistId as per naming convention
5. Use uint64_t for id in test cases, otherwise it will fail
6. Remove extra inclusion of header <random>
